profiles/common/modules/custom/nexteuropa_core_views/nexteuropa_core_views.module

```diff
@@ -11,3 +10,82 @@
 function nexteuropa_core_views_views_api() {
   return ['api' => 3.0];
 }
+
+/**
+ * Implements hook_theme_registry_alter().
+ *
+ * AFAIK, there is no other way to do this horrendous hack to retrieve ALL the
+ * themes hooks available.
+ * We could use the regular way to do it with module_invoke_all('theme') and it
+ * works properly. The only issue is that theme hooks defined in the theme are
+ * not detected because we are using an admin theme.
+ *
+ * Any alternative way is very welcome.
+ */
+function nexteuropa_core_views_theme_registry_alter(&$hook_definitions) {
+  global $theme;
+  // Bails out if we are not in the default theme context.
+  if ($theme != variable_get('theme_default')) {
+    return;
+  }
+
+  $hooks = array();
+  $rows_styles = _nexteuropa_core_views_row_styles();
+  // Reset the variable.
+  variable_set('nexteuropa_core_views_hooks', $hooks);
+
+  foreach ($hook_definitions as $hook => $data) {
+    if (in_array($hook, $rows_styles)) {
+      // Include a file if the theme function or variable processor is held
+      // elsewhere.
+      if (!empty($data['includes'])) {
+        foreach ($data['includes'] as $include_file) {
+          include_once DRUPAL_ROOT . '/' . $include_file;
+        }
+      }
+
+      $function_name = $theme . '_atomium_theme_' . $hook;
+      if (function_exists($function_name)) {
+        $hooks = drupal_array_merge_deep(
+          (array) call_user_func_array(
+            $function_name, array()
+          ),
+          $hooks
+        );
+      }
+    }
+  }
+
+  // Copy hooks data just in case if we need them later in Views.
+  foreach ($hooks as $hook_name => $hook_definition) {
+    $hooks[$hook_name] += $hook_definitions[$hook_name];
+  }
+
+  // Removes hooks with no or empty 'variables' key.
+  array_filter($hooks, function ($item) {
+    return isset($item['variables']) && empty($item['variables']);
+  });
+
+  // Sort hooks alphabetically.
+  ksort($hooks);
+  // Store hooks in a variable, once again, not the best way.
+  variable_set('nexteuropa_core_views_hooks', $hooks);
+}
+
+/**
+ * Build the list of row style plugins.
+ */
+function _nexteuropa_core_views_row_styles() {
+  $ne_core_views_row_styles = [
+    'banner',
+    'card',
+    'blockquote',
+    'timelines',
+    'expandable',
+  ];
+  // Calling all modules implementing hook_my_data_alter():
+  drupal_alter('ne_core_views_row_styles', $ne_core_views_row_styles);
```

Comment:
First, I think we shouldn't shortcut the name nexteuropa to ne.

Second, I would create a new hook: `nexteuropa_core_views_row_styles_info()`. This hook will return the array list, just like you do now.

Then, in `nexteuropa_core_views_theme_registry_alter()`, I'd call a [module_invoke](https://api.drupal.org/api/drupal/includes%21module.inc/function/module_invoke/7.x) on that hook to retrieve them all.

I think it's cleaner and more "Drupalish" :-)